soaks/bin/build_container.sh

```diff
@@ -33,7 +33,7 @@ build_vector() {
         popd
     fi
 
-    docker build --file "${SOAK_ROOT}/Dockerfile" --tag "${IMAGE}" "${BUILD_DIR}"
+    docker build --file "${BUILD_DIR}/soaks/Dockerfile" --tag "${IMAGE}" "${BUILD_DIR}"
```

Comment:
When I went to add `entrypoint.sh` in this PR it failed to build the baseline image because it uses the Dockerfile from this PR, but the baseline SHA as the build context, meaning it lacked `entrypoint.sh`. I think it is more accurate to use the Dockerfile from whatever SHA is being soaked, but open to thoughts here.

Reply:
Ah ha, yes, this is a bug. Nice catch. Our ambition is to _only_ vary the Vector binary in comparison but we were accidentally varying both Vector-the-binary and whatever is in this Dockerfile.

Reply:
Oh, er, wait. Looking at d1a51c3 maybe I'm confused. We should only use the 'comparison' version of the Dockerfile for building both Vector binaries. I believe your change to `build_container` accomplishes that? But d1a51c3 suggests otherwise. 

If we allow for the `Dockerfile` to vary between variants it mostly won't matter since we rarely change these, but it is another variable.

Reply:
Moving conversation to #10978